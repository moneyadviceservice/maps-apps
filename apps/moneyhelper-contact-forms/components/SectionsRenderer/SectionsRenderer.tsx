import { Heading, ListElement } from '@maps-digital/shared/ui';

import { Paragraph } from '@maps-react/common/components/Paragraph';
import { Markdown } from '@maps-react/vendor/components/Markdown';

export type SectionsRendererProps = {
  title?: string;
  content?: string;
  items?: string[];
  footer?: string;
};

export const SectionsRenderer = ({
  sections,
  testIdPrefix,
}: {
  sections: SectionsRendererProps[];
  testIdPrefix: string;
}) => (
  <>
    {sections.map((section, idx) => (
      <section key={idx} className="flex flex-col gap-4">
        {section.title && (
          <Heading
            level="h3"
            component="h2"
            data-testid={`components-${testIdPrefix}-title-${idx}`}
          >
            {section.title}
          </Heading>
        )}
        {section.content && (
          <Markdown content={section.content} className="mb-2" />
        )}
        {section.items && (
          <ListElement
            items={section.items.map((item: string) => (
              <Markdown key={item.slice(0, 8)} content={item} />
            ))}
            variant="unordered"
            color="blue"
            className="mb-3 ml-8"
            data-testid={`${testIdPrefix}-list-${idx}`}
          />
        )}
        {section.footer && (
          <Paragraph testId={`components-${testIdPrefix}-footer-${idx}`}>
            {section.footer}
          </Paragraph>
        )}
      </section>
    ))}
  </>
);
