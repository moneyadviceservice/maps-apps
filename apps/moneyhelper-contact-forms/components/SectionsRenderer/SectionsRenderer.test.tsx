import { render } from '@testing-library/react';

import { SectionsRenderer, SectionsRendererProps } from './SectionsRenderer';

import '@testing-library/jest-dom';

describe('SectionsRenderer', () => {
  const sections: SectionsRendererProps[] = [
    {
      title: 'mock-title',
      content: 'mock-content',
      items: ['mock-item'],
      footer: 'mock-footer',
    },
    {
      title: 'mock-title',
      content: 'mock-content',
      items: ['mock-item', 'mock-item', 'mock-item'],
    },
    {
      content: 'mock-content',
    },
  ];

  const mockTestIdPrefix = 'test-prefix';

  it('matches snapshot for all sections', () => {
    const { container } = render(
      <SectionsRenderer sections={sections} testIdPrefix={mockTestIdPrefix} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders nothing if sections is empty', () => {
    const { container } = render(
      <SectionsRenderer sections={[]} testIdPrefix={mockTestIdPrefix} />,
    );
    expect(container).toBeEmptyDOMElement();
  });
});
