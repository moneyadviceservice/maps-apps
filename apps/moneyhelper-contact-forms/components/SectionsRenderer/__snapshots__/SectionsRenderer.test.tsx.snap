// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionsRenderer matches snapshot for all sections 1`] = `
<div>
  <section
    class="flex flex-col gap-4"
  >
    <h2
      class="text-2xl md:text-4xl text-gray-800 font-bold"
      data-testid="components-test-prefix-title-0"
    >
      mock-title
    </h2>
    <p
      class="mb-2"
      data-testid="paragraph"
    >
      mock-content
    </p>
    <div
      class="flex items-center"
      data-testid="list-element"
    >
      <ul
        class="marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug list-disc marker:text-blue-700 mb-3 ml-8"
        data-testid="test-prefix-list-0"
        start="-1"
      >
        <li>
          <p
            class="mb-4"
            data-testid="paragraph"
          >
            mock-item
          </p>
        </li>
      </ul>
    </div>
    <p
      class="mb-4"
      data-testid="components-test-prefix-footer-0"
    >
      mock-footer
    </p>
  </section>
  <section
    class="flex flex-col gap-4"
  >
    <h2
      class="text-2xl md:text-4xl text-gray-800 font-bold"
      data-testid="components-test-prefix-title-1"
    >
      mock-title
    </h2>
    <p
      class="mb-2"
      data-testid="paragraph"
    >
      mock-content
    </p>
    <div
      class="flex items-center"
      data-testid="list-element"
    >
      <ul
        class="marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug list-disc marker:text-blue-700 mb-3 ml-8"
        data-testid="test-prefix-list-1"
        start="-1"
      >
        <li>
          <p
            class="mb-4"
            data-testid="paragraph"
          >
            mock-item
          </p>
        </li>
        <li>
          <p
            class="mb-4"
            data-testid="paragraph"
          >
            mock-item
          </p>
        </li>
        <li>
          <p
            class="mb-4"
            data-testid="paragraph"
          >
            mock-item
          </p>
        </li>
      </ul>
    </div>
  </section>
  <section
    class="flex flex-col gap-4"
  >
    <p
      class="mb-2"
      data-testid="paragraph"
    >
      mock-content
    </p>
  </section>
</div>
`;
